src/module_wrap.cc

```diff
@@ -609,11 +793,16 @@ void ModuleWrap::Instantiate(const FunctionCallbackInfo<Value>& args) {
   Local<Context> context = obj->context();
   Local<Module> module = obj->module_.Get(isolate);
   TryCatchScope try_catch(realm->env());
-  USE(module->InstantiateModule(
-      context, ResolveModuleCallback, ResolveSourceCallback));
 
-  // clear resolve cache on instantiate
-  obj->resolve_cache_.clear();
+  {
+    ModuleInstantiationContext instantiation_context;
```

Comment:
Do we still need the context if we don't need the globals? The ResolveCache can now be computed at any given time after the source text module is compiled, it would be simpler to just do that e.g. when the ModuleWrap is constructed, and  in the callbacks, just get the cache from the dependent ModuleWrap.

Reply:
This is added to avoid re-entrance of `ModuleWrap::Instantiate`: https://github.com/nodejs/node/pull/59117#discussion_r2219533624

Reply:
Had a second thought that this could be an over design, since the stack-based globals are removed. Updated!